feat(docs): advanced features for 5.0 (#1180)
emqtt/emq-docs-cn: en_US/advanced/shared-subscriptions.md

                                      [s5] got msg1
```

> Subscribtion strategies can be specified globally or per-group. To specify subscribtion strategy per group, use `broker.shared_subscription_group.$group.strategy` option.

Reviewer: ```suggestion
Subscription strategies can be specified globally or per-group. To specify subscription strategy per group, use `broker.shared_subscription_group.$group.strategy` option.
```

EMQX Broker's shared subscription supports balancing strategy and distribution of Ack configuration:

> ```

Reviewer: ```suggestion
```hocon
```

broker {
  # balancing strategy
  shared_subscription_strategy: random
  
  # Applicable to QoS1 QoS2 messages, when enabled,
  # message will be distributed to another subscriber when selected subscriber is offline
  shared_dispatch_ack_enabled = false
}
```
<!-- TODO 待确认 -->

| Balancing strategy |             Description                                                                                                         |
| :----------------- | :------------------------------------------------------------------------------------------------------------------------------ |
| random             | Select randomly among all subscribers                                                                                           |
| round_robin        | According to the order of subscription                                                                                          |
| sticky             | Always sent to the last selected subscriber                                                                                     |
| hash               | According to the hash value of the publisher ClientID                                                                           |

> | hash_clientid      | According to the hash value of the publisher ClientID                                                                           |

Reviewer: I think [`hash`](https://github.com/emqx/emqx/blob/9de327263589e18c324af3642db07751d289378e/apps/emqx/src/emqx_shared_sub.erl#L77) is deprecated, maybe we could omit it?

Author: ```suggestion
| hash_clientid      | According to the hash value of the publisher ClientID                                                                           |
```

| hash_topic         | According to the hash value of the message topic                                                                                |

> | local              | Sends message to a random subscriber of the node received the message. If no local subscribers present, sends one cluster wide. |

Reviewer: ```suggestion
| local              | Sends message to a random subscriber of the node received the message. If no local subscribers are present, sends to a random one cluster in the cluster. |
```
?
